Polio-732  POLIO-628 Editing OrgUnit belonging in invalid group (because of source version)

olethanh opened the pull request:
Fix root cause by Campaign Scope and in general

## Self proof reading checklist

- [ ] Did I use eslint and black formatters
- [ ] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] Did I add translations
- [ ] My migrations file are included
- [ ] Are there enough tests
## Changes
One of the cause of this was that that the group for Campaign scope in polio were not created with the proper source version. This is now fixed at creation and edition.

We also fix this in the org unit API by allowing other field to be edited if the groups membership didn't change and by bypassing the check for empty group.

Note that we don't fix the fact that the group are not displayed properly in the org unit detail and that if you change the group membership you will remove the org unit from the  campaign scope!

## How to test
0. Have the  `polio` plugin enabled.
1. Create/Edit a campaign and put an org unit in the scope
2. Try to edit the org unit in it's detail page.

olethanh reviewed (changes requested):
![image](https://user-images.githubusercontent.com/82500/126613480-f6a6b490-9e6b-4485-b9e7-52e01d327677.png)
Sadly it doesn't work for the dates  in the Round tabs becasue they are stored in child objects. We probably need to do this recursively

quang-le commented:
> ![image](https://user-images.githubusercontent.com/82500/126613480-f6a6b490-9e6b-4485-b9e7-52e01d327677.png)
> Sadly it doesn't work for the dates in the Round tabs becasue they are stored in child objects. We probably need to do this recursively

Added a fix

https://user-images.githubusercontent.com/38907762/126651320-d3bff18d-a997-43b4-a30f-58bc5930a398.mov

olethanh committed:
POLIO-732 Fix scope group not being on the correct source version

no version was attributed on group which paused problem when we were trying to update the orgunit

olethanh committed:
POLIO-732 Only check if group in same version if group has a version

olethanh committed:
PATH /orgunit fix edit orgunit with existing bad group

olethanh committed:
Fix potential bug thank to mypy

mathvdh reviewed (approved):
Je dois dire que je capte pas tout à cette partie mais rien ne m'a choqué en tout cas ..
